embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/storage/EmbraceStorageManager.kt

```diff
@@ -0,0 +1,57 @@
+package io.embrace.android.embracesdk.storage
+
+import io.embrace.android.embracesdk.injection.CoreModule
+import java.io.File
+import java.io.FilenameFilter
+
+internal class EmbraceStorageManager(
+    private val coreModule: CoreModule
+) : StorageManager {
+
+    override val cacheDirectory: Lazy<File> = lazy {
+        coreModule.context.cacheDir
+    }
+
+    override val filesDirectory: Lazy<File> = lazy {
```

Comment:
The problem with doing a custom getter that tries to create the directory each time is that we could end with some files in the cache dir and some on the files/embrace dir. I think it will probably work even in a situation like that, but I think it would be more consistent if we use the same dir. Now, if the `files/embrace` directory can't be created it will use the cache folder as fallback for all files.